packages/lightning-lsp-common/src/resources/sfdx/typings/lds.d.ts

```diff
@@ -119,24 +120,41 @@ declare module 'lightning/uiRecordApi' {
         objectApiName: string;
     }
 
-    export type FieldValueRepresentationValue = null | boolean | number | string | RecordRepresentation;
+    /**
+     * Contains both the raw and displayable field values for a field in a Record.
+     *
```

Comment:
Might be helpful to link to https://developer.salesforce.com/docs/atlas.en-us.uiapi.meta/uiapi/ui_api_responses_field_value.htm